### PR TITLE
feat: deep document ingest

### LINKED ISSUE #63: As TACHIO, I want to repeatedly scrape documents (Such as an SOW, design brief, meeting notes) with important info about ongoing projects, chunking knowledge and embedding it for later retrieval.
Craft doc importer with automatic updating
Detect any docs.pdw.co/* URL, then scrape the plain text or rich text contents and store it in full text full fidelity, semantically chunked vector knowledge (dynamic chunk size based on semantic analysis of text), and a summary of the full document in our knowledgebase. Store a metadata field/flag that indicates this text/knowledge was authored by our team directly. Store all other relevant metadata such as document title, document table of contents, date created, dates of each update, URL of public page, names of other documents or threads where this URL was mentioned, list of any message threads’ conversation IDs where this URL was mentioned, list of other documents that link to/from this document. Store any image attachments and parse through GPT-4V or equivalent vision engine.
Monitor the URL for future changes into perpetuity and trigger an update of the data as needed (this monitoring could be run as a separate service, e.g., self-hosted open source URL monitor)

### PR DESCRIPTION
**Pull request by ejfox**
Closes #63

### CODE REVIEW

**Comment by curran** [comment]
Status:

- [x] Fetches the page
- [x] Chunks the page
- [ ] Stores the chunks as memories

File: capabilities/ingest.js [hunk 1]
```diff
@@ -17,122 +26,95 @@ async function handleCapabilityMethod(method, args) {
...
-    // Then we want to generate a meta-summary of the document based on all of those memories, and store that in the database as well
+    const { html } = await webpageToHTML(url);
+    const document = convert(html, {
+      wordwrap: 130,
+    });
+
+    const messages = [
+      {
+        role: "user",
+        content: `Can you please write an extremely long and thorough reiteration of the following document:
```
Comment (rajivsinclair) [inline capabilities/ingest.js:63]: https://github.com/room302studio/coachartie/issues/116

File: capabilities/ingest.js [hunk 2]
```diff
@@ -17,122 +26,95 @@ async function handleCapabilityMethod(method, args) {
...
+      wordwrap: 130,
+    });
+
+    const messages = [
+      {
+        role: "user",
+        content: `Can you please write an extremely long and thorough reiteration of the following document: 
+${JSON.stringify(document, null, 2)}
+
+When analyzing this document, your goal is to distill its content into concise, standalone facts, as many as you possibly can. Each fact should encapsulate a key piece of information, complete in itself, and easily understandable without needing further context. Pay special attention to precise details, especially if they involve code or search queries - accuracy in phrasing is crucial here. It's important to include relevant URLs, specific search queries, project IDs that are associated with these facts. Respond ONLY with the facts, do not greet me or confirm the request. Keep your response above 1000 words and below 5000 words, please.
```
Comment (rajivsinclair) [inline capabilities/ingest.js:66]: https://github.com/room302studio/coachartie/issues/116

File: capabilities/ingest.js [hunk 3]
```diff
@@ -17,122 +26,95 @@ async function handleCapabilityMethod(method, args) {
...
+        fact,
+        "capability",
+        url,
+      );
+    });
+
+    const metaSummaryMessages = [
+      {
+        role: "user",
+        content: `Can you please provide a high-level summary of the most important facts in this document:
```
Comment (rajivsinclair) [inline capabilities/ingest.js:95]: https://github.com/room302studio/coachartie/issues/116

**Pull request merged**